src/mscorlib/shared/System/Buffers/OwnedMemory.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime;
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers
+{
+    public abstract class OwnedMemory<T> : IDisposable, IRetainable
+    {
+        protected OwnedMemory() { }
+
+        public abstract int Length { get; }
+
+        public abstract Span<T> AsSpan(int index, int length);
+
+        public virtual Span<T> AsSpan()
```

Comment:
Why does this need to be virtual when the `AsSpan(int, int)` overload is already abstract?

Reply:
I am not sure why we need it. I vaguely recall Krzysztof mentioning a performance penalty.
@KrzysztofCwalina, can you help answer this?

https://github.com/dotnet/corefxlab/pull/1527/files#r114916198

Reply:
Regarding why it is virtual. I am not really sure given both IsDisposed and AsSpan(int, int) are both abstract.

Reply:
The reason it was made virtual is that you can imagine a faster implementation than delegating to Span(int, int). But I just discussed this with @ahsonkhan: we would like to remove the AsSpan(int, int) method, make the AsSpan() abstract, and ask callers who want a section of the ownedmemory to call AsSpan().Slice(index, len)